pkg/services/authn/authnimpl/sync/rbac_sync.go

```diff
@@ -121,11 +122,19 @@ func cloudRolesToAddAndRemove(ident *authn.Identity) ([]string, []string, error)
 		return nil, nil, errInvalidCloudRole.Errorf("invalid role: %s", currentRole)
 	}
 
+	// First, add roles for the current role and track them
+	for _, fixedRole := range fixedCloudRoles[currentRole] {
```

Comment:
nit/optional: Another way to do this would be to change `rolesToAdd` to a `map[string]bool` and only return its keys:

```go
rolesToAdd := make(map[string]bool)
for _, fixedRole := range fixedCloudRoles[currentRole] {
    addedFixedRoles[fixedRole] = true
}

for role, fixedRoles := range fixedCloudRoles {
		for _, fixedRole := range fixedRoles {
			if _, ok := addedFixedRoles[fixedRole]; !ok {
				rolesToRemove = append(rolesToRemove, fixedRole)
			}
		}
}

// We do not need to check rolesToAdd's length anymore. 

return maps.Keys(rolesToAdd), rolesToRemove, nil
```

Reply:
neat, thanks!